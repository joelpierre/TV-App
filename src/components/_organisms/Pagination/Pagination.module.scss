.Pagination {
  width: 100%;
  position: relative;
  padding: 0 0 spacing(8);

  &__container {
    @include max-width-container;
    display: flex;
    flex-wrap: wrap;
    text-align: center;
  }

  &__nav,
  &__today {
    width: 100%;
  }

  &__today {
    margin-bottom: spacing(3);
  }

  &__nav {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;

    @include breakpoint('md') {
      flex-wrap: nowrap;
    }
  }

  &__cta {
    position: relative;
    margin-bottom: spacing(2);
    width: 100%;
    background-color: $color-brand;
    padding: spacing(2) spacing(3);
    color: $color-tint-alpha;
    font-weight: bold;
    transition: background-color .4s;

    @include breakpoint('sm') {
      margin: 0 spacing(2) 0;
      min-width: 300px;
      width: auto;
    }

    @include hover-focus {
      color: $color-tint-alpha;
      text-decoration: none;
      background-color: $color-alpha;
    }
  }
}
