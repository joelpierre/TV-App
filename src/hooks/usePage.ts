import React from 'react';
import { useRouter } from 'next/router';

export interface IUsePage {
  page: string;
  isToday: boolean;
}

const usePage = (): IUsePage => {
  const router = useRouter();

  React.useEffect(() => {
    setState({
      page: router?.query?.page as string,
      isToday
    });
  }, [router?.query]);

  const page = router?.query?.page as string || '0';
  const isToday = !page || page === '0';
  const [state, setState] = React.useState<IUsePage>({
    page,
    isToday
  });

  return state;
};

export default usePage;
