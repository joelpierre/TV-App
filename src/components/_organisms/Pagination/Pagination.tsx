import React from 'react';
import classNames from 'classnames';

import styles from './Pagination.module.scss';
import usePage from '@hooks/usePage';
import { getDateFromPage } from 'common/utils/index';
import { EPageHref } from 'common/types/enums';

interface IPaginationProps {
  className?: string;
}

const Pagination: React.FunctionComponent<IPaginationProps> = ({
  className
}) => {
  const { page } = usePage();
  const { date } = getDateFromPage(page);
  const currentPage = parseInt(page, 10) || 0;
  const nextPage = currentPage + 1;
  const prevPage = currentPage - 1;

  return (
    <article className={classNames(styles.Pagination, className)}>
      <div className={styles.Pagination__container}>
        <div className={styles.Pagination__today}>
          <h3 className={styles.Pagination__heading}>
            Whats on: {date}
          </h3>
        </div>

        <div className={styles.Pagination__nav}>
          <a
            href={`${EPageHref.Home}?page=${prevPage}`}
            className={classNames(
              styles.Pagination__cta,
              styles['Pagination__cta--prev']
            )}
            data-testid="prev-link"
          >
            {'<'} View Previous Day
          </a>

          <a
            href={`${EPageHref.Home}?page=${nextPage}`}
            className={classNames(
              styles.Pagination__cta,
              styles['Pagination__cta--next']
            )}
            data-testid="next-link"
          >
            View Next Day {'>'}
          </a>
        </div>
      </div>
    </article>
  );
};

export default Pagination;
