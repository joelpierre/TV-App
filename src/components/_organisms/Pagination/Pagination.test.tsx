import React from 'react';
import { render, RenderResult } from 'test-utils';

import * as hook from '@hooks/usePage';
import * as router from 'next/router';

import Pagination from './Pagination';

const defaultProps = {
  className: 'Test__className'
};

const setup = (props = {}): RenderResult => {
  const setupProps = { ...defaultProps, ...props };
  return render(<Pagination {...setupProps} />);
};

let useRouterSpy;
let usePageSpy;

describe('<Pagination/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    useRouterSpy = jest.spyOn(router, 'useRouter').mockReturnValueOnce({
      query: {
        page: '0'
      }
    } as unknown as router.NextRouter);
    usePageSpy = jest.spyOn(hook, 'default').mockReturnValueOnce({
      isToday: true,
      page: '0'
    });
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should render next and prev buttons', () => {
    expect(wrapper.getByTestId('next-link')).toBeTruthy();
    expect(wrapper.getByTestId('prev-link')).toBeTruthy();
  });
});
