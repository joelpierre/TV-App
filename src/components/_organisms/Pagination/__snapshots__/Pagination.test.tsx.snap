// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> should match snapshot 1`] = `
<DocumentFragment>
  <article
    class="Pagination Test__className"
  >
    <div
      class="Pagination__container"
    >
      <div
        class="Pagination__today"
      >
        <h3
          class="Pagination__heading"
        >
          Whats on: Sun Oct 25 2020
        </h3>
      </div>
      <div
        class="Pagination__nav"
      >
        <a
          class="Pagination__cta Pagination__cta--prev"
          data-testid="prev-link"
          href="/?page=-1"
        >
          &lt; View Previous Day
        </a>
        <a
          class="Pagination__cta Pagination__cta--next"
          data-testid="next-link"
          href="/?page=1"
        >
          View Next Day &gt;
        </a>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
